auth/token.go

```diff
@@ -95,10 +96,27 @@ func TokenLoop(f TokenFunc, done chan interface{}) {
 	}
 }
 
+func delegateIdentityFromToken(token string) DelegateIdentity {
+	identity, err := ParseJWTIdentity(token)
+	if err != nil {
+		log.WithError(err).Error("Unable to obtain identity from token.")
+		return nil
+	}
+	return identity
+}
+
 func updateToken(token string, expires int64) {
 	cond.L.Lock()
 	currentToken = token
+	currentIdentity = delegateIdentityFromToken(token)
 	expiration = time.Unix(expires, 0)
```

Comment:
I think the global `expiration` should be replaced by referencing the current token's expiration.